MekHQ/src/mekhq/campaign/stratcon/StratconRulesManager.java

```diff
@@ -1903,6 +1903,7 @@ public void handleNewDay(NewDayEvent ev) {
             return;
         }
         boolean isMonday = ev.getCampaign().getLocalDate().getDayOfWeek() == DayOfWeek.MONDAY;
+        boolean isStartOfMonth = ev.getCampaign().getLocalDate().getDayOfMonth() == 1;
```

Comment:
If the functions were to receive the localdate and be made to take up whatever decision they can make, we would eliminate those magical booleans in the arguments of a few function calls, and would also not need to create variables like "isMonday" and "isStartOfMonth", because the functions that care about what day it is would be the ones to deal with that kind of trouble.
Since we are not doing those things ate "FPS" scale this should not be such a performance hit for it to worry us.

Reply:
If you want to make a function that does that, it would definitely see use, however checks like this are pretty standard in our codebase and major refactoring is beyond the scope of this PR.